File: src/test/java/TestUtil.java
import org.junit.Assert;
import org.scribble.ext.gt.cli.GTCommandLine2;
//import static org.junit.Assert.assertThrows;

public class TestUtil {

    public static final String BASE_PATH = ".\\src\\test\\java\\oopsla25";

    public static void testGood(String scribFile) {
        GTCommandLine2.main(new String[]{scribFile});
    }

    // TODO GTException
    public static void testBad(String scribFile) {
        // cf. Assertion.assertThrows
        try {
            GTCommandLine2.main(new String[]{scribFile});
        } catch (RuntimeException x) {
            return;
        }
        Assert.fail("Expected exception.");
    }
}

File: src/test/scala/com/github/rhu1/gt/test/oopsla25/TestUtil.scala
package com.github.rhu1.gt.test.oopsla25

import com.github.rhu1.gt.main.Main
import org.scalatest.funsuite.AnyFunSuite;

object TestUtil extends AnyFunSuite {

    val BASE_PATH = "src/test/scala/com/github/rhu1/gt/test/oopsla25"

    def testGood(scribFile: String): Unit =
        Main.main(Array(scribFile, "-gt-check-fidelity", "-gt-check-completeness"))

    def testBad(scribFile: String): Unit =
        assertThrows[Exception](Main.main(Array(scribFile, "-gt-check-fidelity", "-gt-check-completeness")))
}
